tls_codec/src/tls_vec.rs

```diff
@@ -78,6 +78,45 @@ macro_rules! impl_byte_deserialize {
             let result = Self { vec: vec.to_vec() };
             Ok((result, &remainder.get(len..).ok_or(Error::EndOfStream)?))
         }
+
+        fn serialize_bytes_bytes(&$self) -> Result<Vec<u8>, Error> {
```

Comment:
This looks like a lot of duplicate code. Can you factor out some of it to reuse as much as possible?

Reply:
Done.